ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -2223,6 +2237,22 @@ private void getMetaData(QB qb, ReadEntity parentInput)
         }
       }
 
+      Pair<String, String> asOf = qb.getAsOfForAlias(alias);
+      if (asOf != null) {
+        if (!Optional.ofNullable(tab.getStorageHandler()).map(HiveStorageHandler::isTimeTravelAllowed).orElse(false)) {
+          throw new SemanticException(ErrorMsg.TIME_TRAVEL_NOT_ALLOWED, alias);
+        }
+        if (asOf.getLeft() != null) {
+          tab.setAsOfVersion(Long.parseLong(asOf.getLeft()));
+        }
+        if (asOf.getRight() != null) {
+          ZoneId timeZone = SessionState.get() == null ? new HiveConf().getLocalTimeZone() :
+                                 SessionState.get().getConf().getLocalTimeZone();
+          TimestampTZ ts = TimestampTZUtil.parse(stripQuotes(asOf.getRight()), timeZone);
```

Comment:
So this means that the timestamp the user passes into the query will be considered as a timestamp in the user's local time zone (as opposed to a UTC timezone)?

Reply:
Only it the user does not provide the TZ in the string. Same as everywhere in Hive for the `TIMESTAMP WITH LOCAL TIMEZONE` types

Reply:
Understood, thanks